roles/openshift_health_checker/callback_plugins/zz_failure_summary.py

```diff
@@ -82,7 +82,15 @@ def _format_failure(failure):
         (u'Message', stringc(msg, C.COLOR_ERROR)),
     )
     if 'checks' in result._result:
-        rows += ((u'Details', stringc(pformat(result._result['checks']), C.COLOR_ERROR)),)
+        failed_check_msgs = ""
```

Comment:
Suggestions:

- move this logic into a function
- avoid concatenating an arbitrary number of strings (puts pressure on the GC)

```python
def _format_failure(failure):
    # ...
    if 'checks' in result._result:
        rows += (('Failed checks', _format_failed_checks(result._result['checks'])),)
    # ...

def _format_failed_checks(checks):
    output = []
    for check, result in checks.items():
        if not result.get("failed", False):
            continue  # skip checks that did not fail
        output.append(...)
   return stringc("\n".join(output), C.COLOR_ERROR)
```

If you're feeling like refactoring, we could also change `rows` to be a `list`, then `.append` the failed check results into it.